packages/next-swc/crates/next-core/src/app_structure.rs

```diff
@@ -709,9 +721,42 @@ async fn directory_tree_to_entrypoints_internal(
     let subdirectories = &directory_tree.subdirectories;
     let components = directory_tree.components.await?;
 
+    // Capture the current page for the metadata to calculate segment relative to
+    // the corresponding page for the static metadata files.
+    /*
+    let metadata = Metadata {
+        base_page: Some(app_page.clone()),
+        ..components.metadata.clone()
+    }; */
+
+    let components = if components.metadata.base_page.is_some() {
+        components
+    } else {
+        (Components {
+            metadata: Metadata {
+                base_page: Some(app_page.clone()),
+                ..components.metadata.clone()
+            },
+            ..*components
+        })
+        .cell()
+        .await?
+    };
```

Comment:
```suggestion
    let components = if components.metadata.base_page.is_some() {
        Cow::Borrowed(&components)
    } else {
        Cow::Owned(Components {
            metadata: Metadata {
                base_page: Some(app_page.clone()),
                ..components.metadata.clone()
            },
            ..*components
        })
    };
```

No need to cell and read it